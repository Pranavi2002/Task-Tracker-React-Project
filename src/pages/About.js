import React from "react";
import { Box, Typography, useTheme as useMUITheme } from "@mui/material";

export default function About() {
  const muiTheme = useMUITheme();

  return (
    <Box sx={{ p: 3, minHeight: "70vh", bgcolor: "background.default", color: "text.primary" }}>
      <Typography variant="h4" gutterBottom>
        About Page
      </Typography>
      <Typography>
        This is a React Task Tracker app. Features include add/edit/delete tasks, filter, sort, and theme toggle.
      </Typography>
    </Box>
  );
}