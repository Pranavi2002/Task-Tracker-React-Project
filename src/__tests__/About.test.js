// src/__tests__/About.test.js
import React from "react";
import { render } from "@testing-library/react";
import About from "../pages/About";

// Mock useTheme hook
jest.mock("../hooks/useTheme", () => () => ({ theme: "light" }));

describe("About Snapshot", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot(); // Save and check About page UI
  });
});