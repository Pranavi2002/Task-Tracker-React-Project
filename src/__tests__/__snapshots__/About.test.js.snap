// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    style="padding: 20px; min-height: 70vh; background-color: rgb(255, 255, 255); color: rgb(34, 34, 34); transition: 0.3s; text-align: center;"
  >
    <h1>
      About Page
    </h1>
    <p>
      This is a React Task Tracker app. Features include add/edit/delete tasks, filter, sort, and theme toggle.
    </p>
  </div>
</DocumentFragment>
`;
